cmd/oras/internal/errors/errors.go

```diff
@@ -173,3 +173,20 @@ func CheckMutuallyExclusiveFlags(fs *pflag.FlagSet, exclusiveFlagSet ...string)
 	}
 	return nil
 }
+
+// CheckRequiredTogetherFlags checks if any flags required together are all used,
+// returns an error when detecting any flags not used while other flags have been used.
+func CheckRequiredTogetherFlags(fs *pflag.FlagSet, requiredTogetherFlags ...string) error {
+	var unchangedFlags []string
+	for _, flagName := range requiredTogetherFlags {
+		if !fs.Changed(flagName) {
+			unchangedFlags = append(unchangedFlags, fmt.Sprintf("--%s", flagName))
+		}
+	}
+	nUnchangedFlags := len(unchangedFlags)
+	if nUnchangedFlags != 0 && nUnchangedFlags != len(requiredTogetherFlags) {
+		flags := strings.Join(unchangedFlags, ", ")
+		return fmt.Errorf("%s required but not provided", flags)
```

Comment:
The original error is sth like:
> Error: if any flags in the group [cert-file key-file] are set they must all be set; missing [key-file]

Both set and unset flags are all mention, suggestion to follow the design and provide complete error message like:
```suggestion
		return fmt.Errorf("since %s are set, %s required but missing", changed, unchanged)
```